src/monodroid/jni/embedded-assemblies.cc

```diff
@@ -16,77 +16,67 @@ extern "C" {
 #include "monodroid.h"
 #include "dylib-mono.h"
 #include "util.h"
-#include "unzip.h"
-#include "ioapi.h"
 #include "embedded-assemblies.h"
 #include "globals.h"
 #include "monodroid-glue.h"
 
+namespace xamarin { namespace android { namespace internal {
+	struct TypeMappingInfo {
+		char                     *source_apk;
+		char                     *source_entry;
+		int                       num_entries;
+		int                       entry_length;
+		int                       value_offset;
+		const   char             *mapping;
+		TypeMappingInfo          *next;
+	};
+
+	struct md_mmap_info {
+		void   *area;
+		size_t  size;
+	};
+}}}
+
 using namespace xamarin::android;
 using namespace xamarin::android::internal;
 
-struct TypeMappingInfo {
-	char                     *source_apk;
-	char                     *source_entry;
-	int                       num_entries;
-	int                       entry_length;
-	int                       value_offset;
-	const   char             *mapping;
-	struct  TypeMappingInfo  *next;
+const char *EmbeddedAssemblies::suffixes[] = {
+	"",
+	".dll",
+	".exe",
 };
 
-static  MonoBundledAssembly         **bundled_assemblies;
-static  int                           bundled_assemblies_count;
-static  monodroid_should_register     should_register;
-static  void                         *should_register_data;
-static  int                           register_debug_symbols;
-static  char                         *assemblies_prefix;
+constexpr char EmbeddedAssemblies::assemblies_prefix[];
 
-static  struct TypeMappingInfo       *java_to_managed_maps;
-static  struct TypeMappingInfo       *managed_to_java_maps;
 
-MONO_API void
-monodroid_embedded_assemblies_set_should_register (monodroid_should_register r, void *d)
+void EmbeddedAssemblies::set_assemblies_prefix (const char *prefix)
 {
-	should_register       = r;
-	should_register_data  = d;
+	if (assemblies_prefix_override != nullptr)
+		delete[] assemblies_prefix_override;
```

Comment:
This should be `free(assemblies_prefix_override)`, not `delete[]`, as we use `strdup()`.

Reply:
This is in preparation for the [changes in](https://github.com/xamarin/xamarin-android/pull/2611) , it was deliberate